implementations/rust/ockam/ockam_transport_tcp/src/transport_message.rs

```diff
@@ -0,0 +1,120 @@
+use cfg_if::cfg_if;
+use minicbor::{CborLen, Decode, Encode};
+use ockam_core::compat::string::{String, ToString};
+#[cfg(feature = "std")]
+use ockam_core::OpenTelemetryContext;
+#[cfg(feature = "std")]
+use ockam_core::OCKAM_TRACER_NAME;
+use ockam_core::{CowBytes, LocalMessage, Route};
+#[cfg(feature = "std")]
+use opentelemetry::{
+    global,
+    trace::{Link, SpanBuilder, TraceContextExt, Tracer},
+    Context,
+};
+
+/// TCP transport message type.
+#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode, CborLen)]
+#[rustfmt::skip]
+pub struct TcpTransportMessage<'a> {
```

Comment:
It seems to be that most of this code could stay in the `ockam_transport` crate, otherwise we will have to duplicate it for every transport.

It looks like the specificity of TCP is [here](https://github.com/build-trust/ockam/pull/8164/files#diff-8ac45ee8284813729705924bd44521c87c21eb8eeb29fd3385b8bd2ba4d9561cR122), where we serialize the data but that is independent from the `TransportMessage` datatype.

Reply:
I don't see why we can't have different serialization formats for the same, or mostly the same, data types. Now the code above is duplicating [some tracing logic](https://github.com/build-trust/ockam/pull/8164/files#diff-b540e777cb15df1fe532d0c6946f4934d3f201c726d6d0eb5bd2b0ee3ce30dc1R52) that has nothing to do with the serialization.

Reply:
Yes I'd like that.